solace-spring-cloud-stream-binder/solace-spring-cloud-stream-binder-core/src/main/java/com/solace/spring/cloud/stream/binder/SolaceSessionManager.java

```diff
@@ -0,0 +1,158 @@
+      if (this.clientInfoProvider != null) {
+        solaceJcsmpProperties.setProperty(JCSMPProperties.CLIENT_INFO_PROVIDER, clientInfoProvider);
+      }
+
+      try {
+        try {
+          SpringJCSMPFactory springJCSMPFactory = new SpringJCSMPFactory(solaceJcsmpProperties,
+              solaceSessionOAuth2TokenProvider);
+
+          if (solaceSessionEventHandler != null) {
+            LOGGER.debug("Registering Solace Session Event handler on session");
+            context = springJCSMPFactory.createContext(new ContextProperties());
+            jcsmpSession = springJCSMPFactory.createSession(context, solaceSessionEventHandler);
+          } else {
+            jcsmpSession = springJCSMPFactory.createSession();
+          }
+
+          if (solaceSessionEventHandler != null) {
+            solaceSessionEventHandler.setSessionHealthReconnecting();
+          }
+
+          LOGGER.info("Connecting JCSMP session {}", jcsmpSession.getSessionName());
+          jcsmpSession.connect();
+
+          if (solaceSessionEventHandler != null) {
+            solaceSessionEventHandler.setSessionHealthUp();
+          }
+
+          // Check broker compatibility
+          if (jcsmpSession instanceof JCSMPBasicSession session
+              && !session.isRequiredSettlementCapable(Set.of(ACCEPTED, FAILED, REJECTED))) {
+            LOGGER.warn(
+                "The connected Solace PubSub+ Broker is not compatible. It doesn't support message NACK capability. Consumer bindings will fail to start.");
+          }
+
+          LOGGER.info("Successfully created and connected Solace JCSMP session");
+        } catch (JCSMPException exc) {
+          //Throw exception to prevent application from starting if configured to do so (currently the default)
+          if (failOnStartup) {
+            throw exc;
+          }
+        }
+      } catch (Exception e) {
+        LOGGER.error("Failed to initialize Solace JCSMP session", e);
+        if (context != null) {
+          context.destroy();
+        }
+
+        //Throw runtime exception to prevent application from starting
+        throw new SolaceSessionException("Failed to initialize Solace JCSMP session", e);
+      }
+    }
+  }
+
+  public void close() {
+    synchronized (sessionLock) {
+      if (jcsmpSession != null) {
+        try {
+          LOGGER.info("Closing Solace JCSMP session");
+          jcsmpSession.closeSession();
+        } catch (Exception e) {
+          LOGGER.warn("Error closing JCSMP session", e);
+        } finally {
+          jcsmpSession = null;
+        }
+      }
+
+      if (context != null) {
+        try {
+          context.destroy();
+        } catch (Exception e) {
+          LOGGER.warn("Error destroying JCSMP context", e);
+        } finally {
+          context = null;
+        }
+      }
+    }
+  }
+
+  /**
+   * a RuntimeException for Solace session failures.
+   */
+  public static class SolaceSessionException extends RuntimeException {
```

Comment:
do we need a custom exception for this? If not, would it be better to just use a more generic exception?